server/src/main/java/org/apache/iotdb/db/writelog/recover/LogReplayer.java

```diff
@@ -164,11 +164,17 @@ private void replayInsert(InsertPlan plan) throws WriteProcessException, QueryPr
     }
     if (plan instanceof InsertRowPlan) {
       InsertRowPlan tPlan = (InsertRowPlan) plan;
+      //only infer type when users pass a String value
+      //WAL already serializes the real data type, so no need to infer type
+      ((InsertRowPlan) plan).setNeedInferType(false);
       tPlan.setSchemasAndTransferType(schemas);
+      //mark failed plan manually 
+      checkDataTypeAndMarkFailed(schemas, tPlan);
```

Comment:
seems `setSchemasAndTransferType ` has marked the related data point failed already.

Reply:
Hi thank you for your review! 
Here i set plan.setNeedInferType(false)
in setSchemasAndTransferType(schemas), it will only execute this.schemas = schemas
it will not mark the data points as failed

Reply:
I see.